addons/hr_expense/tests/test_expenses_standard_price_update_warning.py

```diff
@@ -54,3 +54,23 @@ def test_expense_standard_price_update_warning(self):
         with Form(self.expense_cat_C, view="hr_expense.product_product_expense_form_view") as form:
             form.standard_price = 5
             self.assertFalse(form.standard_price_update_warning)
+
+    def test_compute_standard_price_update_warning_product_with_and_without_expense(self):
+        self.expense_cat_A = self.env['product.product'].create({
+            'name': 'Category A',
+            'default_code': 'CA',
+            'standard_price': 0.0,
+        })
+        self.expense_cat_B = self.env['product.product'].create({
+            'name': 'Category B',
+            'default_code': 'CB',
+            'standard_price': 0.0,
+        })
+        self.expense_1 = self.env['hr.expense'].create({
+            'employee_id': self.expense_employee.id,
+            'name': 'Expense 1',
+            'product_id': self.expense_cat_A.id,
+            'total_amount': 1,
+        })
+
+        self.env["product.product"].search([["id", "in", (self.expense_cat_A.id, self.expense_cat_B.id)]])._compute_standard_price_update_warning()
```

Comment:
This isn't testing anything

Reply:
because product A has an expense linked to it but not product B.  In the old code, having a recordset containing products with and without a linked expense would cause a traceback because undone_expenses exists, but the key "product B" doesn't exists in mapp.  This is the minimal test that tests that use case in my opinion (although maybe you expect some sort of error assertion?)

Reply:
Indeed, sorry. So first
```suggestion
        })
        
        (self.expense_cat_A |  self.expense_cat_B)._compute_standard_price_update_warning()
```
Then, add a comment and/or a docstring explaining why it would raise

I'd also rename the products:
- expense_cat_A -> product_expensed
- expense_cat_B -> product_not_expensed